stripe-crypto-onramp/src/main/java/com/stripe/android/crypto/onramp/di/OnrampComponent.kt

```diff
@@ -0,0 +1,51 @@
+package com.stripe.android.crypto.onramp.di
+
+import android.app.Application
+import androidx.activity.result.ActivityResultRegistryOwner
+import com.stripe.android.core.injection.CoreCommonModule
+import com.stripe.android.core.injection.CoroutineContextModule
+import com.stripe.android.crypto.onramp.OnrampCoordinator
+import com.stripe.android.crypto.onramp.viewmodels.OnrampCoordinatorViewModel
+import com.stripe.android.paymentelement.callbacks.PaymentElementCallbackIdentifier
+import com.stripe.android.payments.core.injection.StripeRepositoryModule
+import com.stripe.android.ui.core.forms.resources.injection.ResourceRepositoryModule
+import dagger.BindsInstance
+import dagger.Component
+import javax.inject.Singleton
+
+@Singleton
+@Component(
+    modules = [
+        StripeRepositoryModule::class,
+        ResourceRepositoryModule::class,
+        CoreCommonModule::class,
+        CoroutineContextModule::class,
+    ]
+)
+internal interface OnrampComponent {
+    val onrampCoordinator: OnrampCoordinator
+
+    @Component.Builder
+    interface Builder {
+        @BindsInstance
+        fun application(application: Application): Builder
+
+        @BindsInstance
+        fun onRampCoordinatorViewModel(viewModel: OnrampCoordinatorViewModel): Builder
+
+        @BindsInstance
+        fun linkElementCallbackIdentifier(
+            @PaymentElementCallbackIdentifier linkElementCallbackIdentifier: String
+        ): Builder
+
+        @BindsInstance
+        fun activityResultRegistryOwner(
+            activityResultRegistryOwner: ActivityResultRegistryOwner
+        ): Builder
+
+        @BindsInstance
+        fun isLinkUserCallback(isLinkUserCallback: (Boolean) -> Unit): Builder
```

Comment:
can we instead create a class for the for this? you can do it on a separate PR but the expectation would be something like:

```
fun interface OnRampLookupCallback {
      fun onResult(result: OnRampLookupResult)
}


/**
 * Result of an OnRamp Link user lookup operation.
 */
sealed interface OnRampLookupResult {
    /**
     * Link user lookup was successful.
     * @param isLinkUser Whether the email belongs to an existing Link user
     */
    class Completed internal constructor(
        val isLinkUser: Boolean
    ) : OnRampLookupResult

    /**
     * Link user lookup failed.
     * @param error The error that caused the failure
     */
    class Failed internal constructor(
        val error: Throwable
    ) : OnRampLookupResult
}
```

Reply:
Yeah, I'll tackle this along with the other callbacks in the next PR.